openedx/core/djangoapps/discussions/handlers.py

```diff
@@ -0,0 +1,93 @@
+"""
+Signal handlers for discussions events
+"""
+import logging
+from uuid import uuid4
+
+from django.db import transaction
+
+from openedx.core.djangoapps.discussions.data import CourseDiscussionConfigurationData
+from openedx.core.djangoapps.discussions.models import (
+    DEFAULT_PROVIDER_TYPE,
+    DiscussionTopicLink,
+    DiscussionsConfiguration,
+)
+from openedx.core.djangoapps.discussions.signals import COURSE_DISCUSSIONS_UPDATED
+
+log = logging.getLogger(__name__)
+
+
+# pylint: disable=unused-argument
+def handle_course_discussion_config_update(sender, configuration: CourseDiscussionConfigurationData, **kwargs):
+    """
+    Updates the database models for course topics and configuration when settings are updated in the course.
+
+    Args:
+        sender: Ignored
+        configuration (CourseDiscussionConfigurationData): configuration data for the course
+
+    """
+    update_course_discussion_config(configuration)
+
+
+def update_course_discussion_config(configuration: CourseDiscussionConfigurationData):
+    """
+    Update the database version of the configuration if it changes in the course structure.
+    Args:
+        configuration (CourseDiscussionConfigurationData): configuration data for the course
+    """
+    course_key = configuration.course_key
+    provider_id = configuration.provider_type or DEFAULT_PROVIDER_TYPE
+    new_topic_map = {
+        (topic_context.usage_key or topic_context.external_id): topic_context
+        for topic_context in configuration.contexts
+    }
+    with transaction.atomic():
+        # Go over existing topics in the database and update them if they have been changed.
```

Comment:
If we can add a more detailed explanation in the doc string that would make it easier to understand. Would also remove all these comments in the code

Reply:
Sure.